streamz/core.py

```diff
@@ -72,6 +73,14 @@ def __init__(self, child=None, children=None, stream_name=None, **kwargs):
                 child.parents.add(self)
         self.name = stream_name
 
+    @classmethod
+    def register_api(cls, func):
+        @functools.wraps(func)
+        def _(*args, **kwargs):
+            return func(*args, **kwargs)
+        setattr(cls, func.__name__, _)
```

Comment:
Just out of mere curiosity (I can't see the subtle reason as of now), why this step and not just use `func`?

Reply:
Python didn't like it.  I guess it didn't identify it as a method and so didn't pass it `self`.  This gets into levels of Python that I never cared to learn about.

Reply:
thanks!